### PR TITLE
centrifugo: init at 5.1.1

### PR DESCRIPTION
**Pull request by tie**
## Description of changes

Packages [Centrifugo](https://centrifugal.dev), scalable real-time messaging server.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/networking/centrifugo.nix [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    enable = lib.mkEnableOption (lib.mdDoc "Centrifugo messaging server");
+
+    package = lib.mkPackageOptionMD pkgs "centrifugo" { };
+
+    settings = lib.mkOption {
+      default = { };
+      type = format.type;
+      description = lib.mdDoc ''
+        Declarative Centrifugo configuration. See the [Centrifugo
+        documentation] for a list of options.
```
Comment (jian-lin) [inline nixos/modules/services/networking/centrifugo.nix:20]: I have never seen this syntax for a link and it is not listed in the commonmark [website](https://commonmark.org/help/)

**Review by jian-lin — COMMENTED** [review]
The NixOS module looks great!

I do not review the package though.

File: nixos/modules/services/networking/centrifugo.nix (hunk 1 above)
Comment (tie) [reply nixos/modules/services/networking/centrifugo.nix:20]: See https://spec.commonmark.org/0.30/#link-reference-definitions

Comment (tie) [reply nixos/modules/services/networking/centrifugo.nix:20]: This makes reading raw Markdown a bit easier since long URLs are not interleaved with text.

**Comment by tie** [comment]
@jian-lin, can you take a look at this again please, is there anything else that should be done for this package to be merged?

**Comment by jian-lin** [comment]
Sorry, I am not familiar with go packaging, so I cannot help review the package.

**Comment by tie** [comment]
@NickCao, would you mind taking a look at this?

**Review by NickCao — APPROVED** [review]
LGTM

**Pull request merged**